consumer/src/main/kotlin/streams/kafka/KafkaEventSink.kt

```diff
@@ -35,20 +36,12 @@ class KafkaEventSink(private val config: Config,
         return object: StreamsEventConsumerFactory() {
             override fun createStreamsEventConsumer(config: Map<String, String>, log: Log): StreamsEventConsumer {
                 val kafkaConfig = KafkaSinkConfiguration.from(config)
-                val dlqService = if (kafkaConfig.streamsSinkConfiguration.dlqTopic.isNotBlank()) {
-                    val asProperties = kafkaConfig.asProperties()
-                            .mapKeys { it.key.toString() }
-                            .toMutableMap()
-                    asProperties.remove(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG)
-                    asProperties.remove(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG)
-                    KafkaDLQService(asProperties, "__streams.errors")
-                } else {
-                    null
-                }
+
+                val errorService = KafkaDLQService(kafkaConfig.asProperties(), ErrorService.ErrorConfig.from(kafkaConfig.streamsSinkConfiguration.errorConfig),{ s, e -> log.error(s,e as Throwable)})
```

Comment:
Couple of comments here --
please break up this line, it's long, complicated, nested, I found it pretty tough to read.
Lastly that final argument:

```
{ s, e -> log.error(s,e as Throwable)
```

You initialize KafkaDLQService differently in another place, with a different final argument function (similar but still).  For consistency you might remove this argument altogether, or have it start with a default value so you can be consistent everywhere.   Within the StreamsEventConsumerFactory, I don't think it's this object's responsibility to specify error logging behavior.  Because separate code elsewhere deals with all of the config values, I think that's probably the right place to set this.

Reply:
no we need the argument for logging,
it's because different logger implementations (connect-plugin vs. neo4j extension) have different signatures, and here, unfortunately the signature cannot be resolved with  a method references as it's ambiguous.

Reply:
please leave a comment here on why then.  That makes sense but I would have missed that going through the code here, because the code looks trivial and refactorable